src/basic/fd-util.h

```diff
@@ -121,3 +125,21 @@ static inline char *format_proc_fd_path(char buf[static PROC_FD_PATH_MAX], int f
 
 #define FORMAT_PROC_FD_PATH(fd) \
         format_proc_fd_path((char[PROC_FD_PATH_MAX]) {}, (fd))
+
+/* Determining a file descriptor's path via /proc is not reliable. Only use this function to provide better
+ * log messages when doing fd based path handling! */
+static inline const char *FORMAT_FD_PATH(int fd, char **buf) {
```

Comment:
Why inline? We usually do not introduce such long function in inline form.
Also, we usually takes buffer as a argument, i.e. `char *buf, size_t len`, then introduce a macro with a compound literal.

Reply:
This was discussed quite extensively in the other PR, we cannot use stack allocated buffers here because we would end up consuming way too much stack space.

Reply:
Oh, I did not know that. Could you provide the reference?

Reply:
https://github.com/systemd/systemd/pull/25836#discussion_r1056398315

Reply:
Thanks. Yeah, PATH_MAX is quite large...

Reply:
But, still I prefer that this function in a non-inline form, and name in lower characters.